files/certbot_lock.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+LOCKFILE="/var/lock/certbot"
+LOCKFD=99
+# private
+function _lock()          { flock "$@" $LOCKFD; }
```

Comment:
`flock(1)` does not appear to be something standard. On FreeBSD for example, we have `lockf(1)` (which seems to serve the same purpose) but no `flock(1)`.

The whole logic seems over-engineered, `flock(1)` and `lockf(1)` both seems to support a simple syntax with a file and a command where the file is locked while the command run. Maybe you can try to find if the system has `lockf(1)` / `flock(1)` (e.g. using the `type` built-in command) and use it if found; and otherwise fallback to not locking?

Reply:
On Debian `flock` is part of `util-linux` which is [essential package](https://packages.debian.org/bullseye/util-linux) that is automatically installed. But of course other distributions might take very different approach.

I've tried to keep the code as simple as possible, but if you have better suggestion, feel free to improve it.

Reply:
set -o noclobber not enough?

Reply:
@vchepkov No, it's not enough. We don't want immediate error (that's already covered by certbot), we need lock & wait for releasing previous lock (if exists).

@smortex I'm not familiar with FreeBSD, but in FreeBSD 13.8 `flock` [seems to be available](https://man.freebsd.org/cgi/man.cgi?query=flock&sektion=1&manpath=FreeBSD+13.2-RELEASE+and+Ports). `lockf` is only a C function (same on Linux).